File: bioinformatics-stronghold/rna.py
"""
Problem ID: RNA
http://rosalind.info/problems/rna/

Transcribing DNA into RNA
"""

def transcribe_dna_to_rna(s):
    return s.strip().replace('T', 'U')


s = 'GATGGAACTTGACTACGTAAATT'
assert(transcribe_dna_to_rna(s) == 'GAUGGAACUUGACUACGUAAAUU')


with open('data/rosalind_rna.txt', 'r') as f:
    print(transcribe_dna_to_rna(f.read()))

File: bioinformatics-stronghold/rna.scala
// Problem ID: RNA
// Transcribing DNA to RNA

import scala.io.Source

def transcribeRNAtoDNA(s: String): String = {
    s.trim.replace('T', 'U')
}

val s = "GATGGAACTTGACTACGTAAATT"
assert(transcribeRNAtoDNA(s) == "GAUGGAACUUGACUACGUAAAUU")

var line = Source.fromFile("data/rosalind_rna.txt").mkString
println(transcribeRNAtoDNA(line))